mutatest/run.py

```diff
@@ -382,9 +373,16 @@ def create_mutation_run_trial(
 
     LOGGER.debug("Running trial for %s", mutation_op)
     mutant = genome.mutate(target_idx, mutation_op, write_cache=True)
-    mutant_trial = subprocess.run(
-        test_cmds, capture_output=capture_output(LOGGER.getEffectiveLevel())
-    )
+    try:
+        mutant_trial = subprocess.run(
+            test_cmds,
+            capture_output=capture_output(LOGGER.getEffectiveLevel()),
+            timeout=max_runtime,
+        )
+    except subprocess.TimeoutExpired:
+        cache.remove_existing_cache_files(mutant.src_file)
+        return MutantTrialResult(mutant=mutant, return_code=3)
```

Comment:
This logic makes sense to me, in a refactor I think we can remove the `cache` and `return` statements from the `except` block and replace it with an alternate flag for the `return_code` (using a variable set to None earlier in the function). For example:

```python
return_code=None
try:
   mutant_trials = ...
except subprocess.TimeoutExpired:
   return_code = 3

cache.remove_existing ...
return MutantTrialResults(
    mutant=mutant,return_code=(return_code or mutant_trial.returncode)
)
```

That should simplify testing later since the exception only sets the timeout returncode and other core logic is not duplicated.